Automattic/wp-calypso
# Add link to the customize the header to the page preview

#### Changes proposed in this Pull Request

* Add an optional link to the customizer to customize the header.
* Default it to be hidden
* Hide it in the site preview mode ( there is no room for it because the domain name is visible in the middle of the header)
* Show it only within the block editor

#### Testing instructions

1) Edit a page
    * Open the preview dialog
    * The Edit Header button should be visible and take you to edit the site header in the customizer
![Screen Shot 2020-05-07 at 10 27 59 AM](https://user-images.githubusercontent.com/22446385/81241432-941c8500-904d-11ea-864c-540431e06f1c.png)

2) Click on your site name in the top left corner of the wordpress menu e.g. go to
/view/{siteSlug}
    * The Edit Header button should not be visible
    * The domain name of the site should be visible in the middle of the preview header.

![Screen Shot 2020-05-07 at 10 28 10 AM](https://user-images.githubusercontent.com/22446385/81241426-9088fe00-904d-11ea-8d37-6e0f50cf0a8e.png)

Fixes #41295
